packages/flutter_tools/lib/executable.dart

```diff
@@ -38,8 +38,7 @@ import 'src/runner/flutter_command.dart';
 Future<Null> main(List<String> args) async {
   final bool verbose = args.contains('-v') || args.contains('--verbose');
 
-  final bool doctor = (args.isNotEmpty && args.first == 'doctor') ||
-      (args.length == 2 && verbose && args.last == 'doctor');
+  final bool doctor = args.contains('doctor');
```

Comment:
I think this is too broad;  any parameter named "doctor" will trigger the flutter doctor here.

Reply:
as an example:  "flutter create doctor" will have the side effect of muting the command logging, which is probably not what you want.

Reply:
Hmm, we could explicitly check for `doctor`, `-v doctor`, and `doctor -v`.